compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -292,6 +298,86 @@ impl<'hir> LoweringContext<'_, 'hir> {
         }
     }
 
+    /// Checks if an expression refers to a function marked with
+    /// `#[rustc_legacy_const_generics]` and returns the argument index list
+    /// from the attribute.
+    fn legacy_const_generic_args(&mut self, expr: &Expr) -> Option<Vec<usize>> {
+        if let ExprKind::Path(None, path) = &expr.kind {
+            if path.segments.last().unwrap().args.is_some() {
+                return None;
+            }
```

Comment:
since we don't have any such functions right now, the legacy support doesn't need to support it. If anyone creates a function like this, they'll just have to use const generics syntax. This PR is just so the call sites to existing functions keep working